### PR TITLE
VPC API compatibility fix

### PR DESCRIPTION
**Pull request by sameshai**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,6 +1,6 @@
 #FROM alpine:3.7
 ARG PROXY_IMAGE_URL=blank
-FROM ubuntu:18.04
+FROM ubuntu:16.04
```
Comment (prankulmahajan) [inline Dockerfile:3]: Any reason why is this reverted to ubuntu 16?

Comment (prankulmahajan) [inline Dockerfile:3]: Any reason why is this reverted to ubuntu 16?

File: pkg/ibmcsidriver/controller_helper.go [hunk 2]
```diff
@@ -50,6 +50,20 @@ var customCapacityIopsRanges = []classRange{
 	{1000, 1999, 100, 20000},
 }
 
+// Range as per IBM volume provider Storage for DP2 Profile
```
Comment (prankulmahajan) [inline pkg/ibmcsidriver/controller_helper.go:53]: Can you give the ref of the public doc in comment?

File: pkg/ibmcsidriver/controller_helper.go [hunk 3]
```diff
@@ -360,14 +384,14 @@ func overrideParams(logger *zap.Logger, req *csi.CreateVolumeRequest, config *co
 			}
 		case IOPS:
 			// Override IOPS only for custom class
```
Comment (prankulmahajan) [inline pkg/ibmcsidriver/controller_helper.go:386]: "Override IOPS only if the profile if either custom or dp2"

**Review by prankulmahajan — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 1 above)
Comment (sameshai) [reply Dockerfile:3]: Travis is failing it is not able to pull 18 or any other ubuntu image. Some travis and distro specific issue. Anyways we are not going to use this for managed env

**Review by prankulmahajan — DISMISSED** [review]
lgtm

**Comment by prankulmahajan** [comment]
@sameshai Can you add the new dp2 storage Classes in this repo?

**Comment by sameshai** [comment]
> @sameshai Can you add the new dp2 storage Classes in this repo?

done

File: deploy/kubernetes/storageclass/ibmc-vpc-file-dp2-StorageClass.yaml [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  profile: "dp2"            # The VPC Storage profile used. https://cloud.ibm.com/docs/vpc?topic=vpc-file-storage-profiles&interface=ui#dp2-profile
+  iops: "100"                       # Default IOPS. User can override from secrets
+  billingType: "hourly"             # The default billing policy used. User can override this default
+  encrypted: "false"                # By default, all PVC using this class will only be provider managed encrypted. The user can override this default
+  encryptionKey: ""                 # If encrypted is true, then a user must specify the encryption key used associated KP instance
+  resourceGroup: ""                 # Use resource group if specified here. else use the one mentioned in storage-secrete-store
+  zone: ""                          # By default, the storage vpc driver will select a zone. The user can override this default
+  tags: ""                          # A list of tags "a, b, c" that will be created when the volume is created. This can be overidden by user
+  classVersion: "1"
+  UID: "0"                           # The initial user identifier for the file share.
```
Comment (prankulmahajan) [inline deploy/kubernetes/storageclass/ibmc-vpc-file-dp2-StorageClass.yaml:20]: This is lower case

File: deploy/kubernetes/storageclass/ibmc-vpc-file-dp2-StorageClass.yaml [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+  GID: "0"                           # The initial group identifier for the file share.
```
Comment (prankulmahajan) [inline deploy/kubernetes/storageclass/ibmc-vpc-file-dp2-StorageClass.yaml:21]: This is lower case too

Comment (sameshai) [reply deploy/kubernetes/storageclass/ibmc-vpc-file-dp2-StorageClass.yaml:21]: You did introduce this in custom storage class :)

File: pkg/ibmcsidriver/node.go [hunk 6]
```diff
@@ -75,14 +74,6 @@ func (su *VolumeStatUtils) FSInfo(path string) (int64, int64, int64, int64, int6
...
 
 const (
-	// DefaultVolumesPerNode is the default number of volumes attachable to a node
-	DefaultVolumesPerNode = 4
-
-	// MaxVolumesPerNode is the maximum number of volumes attachable to a node
-	MaxVolumesPerNode = 12
-
-	// MinimumCoresWithMaximumAttachableVolumes is the minimum cores required to have maximum number of attachable volumes, currently 4 as per the docs.
-	MinimumCoresWithMaximumAttachableVolumes = 4
```
Comment (prankulmahajan) [inline pkg/ibmcsidriver/node.go:85]: Can you please add the issue and the tests that you ran after this

**Review by prankulmahajan — APPROVED** [review]
lgtm

File: pkg/ibmcsidriver/node.go (hunk 6 above)
Comment (sameshai) [reply pkg/ibmcsidriver/node.go:85]: **Issue was there was limit of 12 volume max per node**

```

Events:
  Type     Reason            Age                    From               Message
  ----     ------            ----                   ----               -------
  Warning  FailedScheduling  9m26s                  default-scheduler  0/1 nodes are available: pod has unbound immediate PersistentVolumeClaims. preemption: 0/1 nodes are available: 1 No preemption victims found for incoming pod..
  Warning  FailedScheduling  3m27s                  default-scheduler  0/1 nodes are available: 1 node(s) exceed max volume count. preemption: 0/1 nodes are available: 1 No preemption victims found for incoming pod..
  Warning  FailedScheduling  8m52s (x4 over 9m25s)  default-scheduler  0/1 nodes are available: pod has unbound immediate PersistentVolumeClaims. preemption: 0/1 nodes are available: 1 No preemption victims found for incoming pod..
```

**Fix to remove this check**

```
root@test-ciocmnc205mphmigkus0-samclustert-default-00000347:/# df -h | grep fsf      
fsf-dal1095b-vs91-stg-fz.adn.networklayer.com:/1d304568_edda_44bf_85ee_e74c5931a3ed  10G 320K  10G  1% /var/data/kubelet/pods/d51faf52-4a4f-4f78-8c85-7d346cb86e11/volumes/kubernetes.io~csi/pvc-967c6448-47aa-4931-86a8-ad773f032b56/mount
fsf-dal1095b-vs91-stg-fz.adn.networklayer.com:/5cfc71e2_b209_46b6_bfad_43d7d6cc55be  10G 320K  10G  1% /var/data/kubelet/pods/1c8101e5-3c6d-4626-97ff-40619123ed48/volumes/kubernetes.io~csi/pvc-d1bc3e37-9560-4822-8038-235e0acd2d52/mount
fsf-dal1095a-vs91-stg-fz.adn.networklayer.com:/e489966e_90fd_4f12_9dda_2d78fe319e37  10G 320K  10G  1% /var/data/kubelet/pods/b41cb7a7-8c5f-4dd9-9e61-18a778356193/volumes/kubernetes.io~csi/pvc-91d93513-039a-46cd-a3f5-018b99d66006/mount
fsf-dal1095a-vs91-stg-fz.adn.networklayer.com:/a6b51e31_b07b_481f_b47b_be1a47a9dcaa  10G 320K  10G  1% /var/data/kubelet/pods/3ec2a6ec-9046-4a03-9d8f-6354b49ae44e/volumes/kubernetes.io~csi/pvc-483ebbd1-9655-4fd9-85d1-743c94481d48/mount
fsf-dal1095b-vs91-stg-fz.adn.networklayer.com:/38d1ec7d_30af_43fc_8ec2_91dcf4b1131d  10G 320K  10G  1% /var/data/kubelet/pods/7ce64836-2fc1-4c20-bd3a-d8d44a55c752/volumes/kubernetes.io~csi/pvc-31d55979-b4f1-420d-9234-21d650347c0e/mount
fsf-dal1095b-vs91-stg-fz.adn.networklayer.com:/53ca5e04_567b_4486_8db4_ac7c6c816ba9  10G 320K  10G  1% /var/data/kubelet/pods/75776372-f867-46b6-956d-c559e1effab4/volumes/kubernetes.io~csi/pvc-377f119b-6629-40b1-a931-b0761a941aa9/mount
fsf-dal1095b-vs91-stg-fz.adn.networklayer.com:/12903b77_46e5_48c4_b96d_773244fb0185  10G 320K  10G  1% /var/data/kubelet/pods/d3fc3fc0-9e75-4877-a964-0382040c4ca1/volumes/kubernetes.io~csi/pvc-7af59942-21eb-49f0-a5c2-4da6982de06d/mount
fsf-dal1095b-vs91-stg-fz.adn.networklayer.com:/6488c4db_9f35_4e54_8ebb_45d3926bae36  10G 320K  10G  1% /var/data/kubelet/pods/39b6a6c2-3225-485f-88f3-1e2224fe6fc1/volumes/kubernetes.io~csi/pvc-49425522-1130-44b1-9a00-43089aefba3b/mount
fsf-dal1095a-vs91-stg-fz.adn.networklayer.com:/f82231f6_3a84_4520_94bc_4a6a557a2be9  10G 320K  10G  1% /var/data/kubelet/pods/ff244a03-fd81-4909-9a1f-1bdbb8f8dcba/volumes/kubernetes.io~csi/pvc-1eb8ee82-5e1b-4fb7-b04c-003eb3534952/mount
fsf-dal1095b-vs91-stg-fz.adn.networklayer.com:/2bf6ef2a_3826_4b1f_a448_cd728460eada  10G 320K  10G  1% /var/data/kubelet/pods/84fdf724-733c-41bf-b3f8-75d868931550/volumes/kubernetes.io~csi/pvc-75805ff5-c42f-4edc-a29a-bca760fd8573/mount
fsf-dal1095b-vs91-stg-fz.adn.networklayer.com:/26ad45b6_1ad1_4ae6_825b_27cbd3ef1f74  10G 320K  10G  1% /var/data/kubelet/pods/bb6019ce-5413-4da8-950a-2aa12d2766e6/volumes/kubernetes.io~csi/pvc-bfde2627-c99c-44e3-895c-9e10d57c40a9/mount
fsf-dal1095b-vs91-stg-fz.adn.networklayer.com:/f60ec382_9faf_4e10_b61e_bee2c8c5c74f  10G 320K  10G  1% /var/data/kubelet/pods/ebef3798-6526-4e17-8e70-c5be07626b5e/volumes/kubernetes.io~csi/pvc-53432118-59d8-47b4-b5db-5912ce32ba62/mount
fsf-dal1095a-vs91-stg-fz.adn.networklayer.com:/67281fb1_d5a8_4409_bb94_41602818cb3e  10G 320K  10G  1% /var/data/kubelet/pods/3f554844-9744-42d4-b389-823909c79624/volumes/kubernetes.io~csi/pvc-a87c1265-76c2-4745-a770-77815fbf5ffa/mount
fsf-dal1095b-vs91-stg-fz.adn.networklayer.com:/56f1d7f2_9675_441a_8874_26b44f893144  10G 320K  10G  1% /var/data/kubelet/pods/f3ce0298-c835-4e27-b149-a996428f759d/volumes/kubernetes.io~csi/pvc-31433d37-0855-4050-973f-e987379abca7/mount
fsf-dal1095b-vs91-stg-fz.adn.networklayer.com:/e9f637ca_1e5b_4384_90fa_f34c77fa86b8  10G 320K  10G  1% /var/data/kubelet/pods/ecbca4dd-1320-4016-8bdc-0893e7f7f1f4/volumes/kubernetes.io~csi/pvc-ee119518-c7c9-4324-9995-c69259a5a3e1/mount

NAME               READY  STATUS  RESTARTS  AGE   IP       NODE      NOMINATED NODE  READINESS GATES
deployment-1-79f7b98b6-c2sxg   1/1   Running  0     5m13s  172.17.47.29  10.240.128.4  <none>      <none>
deployment-10-57458db988-x6cc6  1/1   Running  0     4m39s  172.17.59.106  10.240.0.4   <none>      <none>
deployment-11-5569d584b6-xv9gt  1/1   Running  0     4m36s  172.17.47.15  10.240.128.4  <none>      <none>
deployment-12-68cbcf78bd-qlxzr  1/1   Running  0     4m32s  172.17.59.107  10.240.0.4   <none>      <none>
deployment-13-98778c68b-gwk7m  1/1   Running  0     4m28s  172.17.47.16  10.240.128.4  <none>      <none>
deployment-14-d4b777684-p2v4x  1/1   Running  0     4m25s  172.17.59.108  10.240.0.4   <none>      <none>
deployment-15-6cc7d579d6-d65hz  1/1   Running  0     4m20s  172.17.47.34  10.240.128.4  <none>      <none>
deployment-16-6b6f8965c6-kjkw9  1/1   Running  0     4m16s  172.17.59.109  10.240.0.4   <none>      <none>
deployment-17-7f4d7f464b-tqhvf  1/1   Running  0     4m12s  172.17.47.35  10.240.128.4  <none>      <none>
deployment-18-7f7966ddd4-5ml6l  1/1   Running  0     4m8s  172.17.59.110  10.240.0.4   <none>      <none>
deployment-19-59dff494c4-6vz8c  1/1   Running  0     4m4s  172.17.47.38  10.240.128.4  <none>      <none>
deployment-2-74f6c954c4-8sz5x  1/1   Running  0     5m9s  172.17.59.102  10.240.0.4   <none>      <none>
deployment-20-55864b6df4-ng9ws  1/1   Running  0     4m1s  172.17.59.111  10.240.0.4   <none>      <none>
deployment-21-76f4b48bb8-snjwn  1/1   Running  0     3m56s  172.17.47.39  10.240.128.4  <none>      <none>
deployment-22-55d89d64b-nzqg2  1/1   Running  0     3m52s  172.17.59.112  10.240.0.4   <none>      <none>
deployment-23-7b988b6bd4-9srbp  1/1   Running  0     3m49s  172.17.47.41  10.240.128.4  <none>      <none>
deployment-24-7bcd7f8fd-629zb  1/1   Running  0     3m45s  172.17.59.114  10.240.0.4   <none>      <none>
deployment-25-547bff449b-kgl6b  1/1   Running  0     3m41s  172.17.47.40  10.240.128.4  <none>      <none>
deployment-26-68576db578-znbmc  1/1   Running  0     3m37s  172.17.59.113  10.240.0.4   <none>      <none>
deployment-27-7864995684-v7ckb  1/1   Running  0     3m33s  172.17.47.50  10.240.128.4  <none>      <none>
deployment-28-6485ff5b54-n8fnp  1/1   Running  0     3m29s  172.17.59.115  10.240.0.4   <none>      <none>
deployment-29-7c6dff7764-zd68p  1/1   Running  0     3m25s  172.17.47.46  10.240.128.4  <none>      <none>
deployment-3-c58bff87d-g2wsv   1/1   Running  0     5m5s  172.17.47.30  10.240.128.4  <none>      <none>
deployment-30-5c7d87549b-xjd4d  1/1   Running  0     3m21s  172.17.59.116  10.240.0.4   <none>      <none>
deployment-4-569f46ddf8-6xr28  1/1   Running  0     5m2s  172.17.59.103  10.240.0.4   <none>      <none>
deployment-5-56cdfdfd66-rt6ck  1/1   Running  0     4m58s  172.17.47.31  10.240.128.4  <none>      <none>
deployment-6-65b67794f8-lbpcz  1/1   Running  0     4m54s  172.17.59.104  10.240.0.4   <none>      <none>
deployment-7-7c5db8d7c8-7wr8r  1/1   Running  0     4m51s  172.17.47.32  10.240.128.4  <none>      <none>
deployment-8-6d6687d674-rsqgn  1/1   Running  0     4m47s  172.17.59.105  10.240.0.4   <none>      <none>
deployment-9-678bfc58cd-wz78w  1/1   Running  0     4m43s  172.17.47.33  10.240.128.4  <none>      <none>
```

**Review by arahamad — APPROVED** [review]
lgtm

**Pull request merged**